src/Hazelcast.Net/Clustering/ClusterMembers.cs

```diff
@@ -167,18 +176,25 @@ public void AddConnection(MemberConnection connection, bool isNewCluster)
                     if (!_connected)
                     {
                         // change Started | Disconnected -> Connected, ignore otherwise, it could be ShuttingDown or Shutdown
-                        _logger.LogDebug($"Added connection {connection.Id.ToShortString()} to member {connection.MemberId.ToShortString()}, now connected.");
+                        _logger.LogDebug($"Added connection {connection.Id.ToShortString()} to member {connection.MemberId.ToShortString()} at {connection.Address}, now connected.");
                         _clusterState.ChangeState(ClientState.Connected, ClientState.Started, ClientState.Disconnected);
                         _connected = true;
                     }
                     else
                     {
-                        _logger.LogDebug($"Added connection {connection.Id.ToShortString()} to member {connection.MemberId.ToShortString()}.");
+                        if (_logger.IsEnabled(LogLevel.Debug))
+                        {
+                            var msg = $"Added connection {connection.Id.ToShortString()} to member {connection.MemberId.ToShortString()} at {connection.Address}";
+                            msg += existingConnection == null
+                                ? "."
+                                : $", replacing connection {existingConnection.Id.ToShortString()} at {existingConnection.Address}.";
+                            _logger.LogDebug(msg);
+                        }
```

Comment:
Should we log something here when the LogLevel is not debug?
If it is unneccessary, what about the using `else if` directly